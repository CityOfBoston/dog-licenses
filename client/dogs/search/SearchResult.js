// @flow

import React from 'react';
import Link from 'next/link';

import type { DogLicense } from '../../types';
import { GRAY_100 } from '../../common/style-constants';

export type Props = {|
  license: DogLicense,
|};

export default function SearchResult({
  license: { firstName, lastName, dogName, id, phone },
}: Props) {
  return (
    <Link href={`/dogs/license?id=${id}`} as={`/dogs/license/${id}`}>
      <a className={'p-a300 br br-t100 b--w result'}>
        <div
          className="t--sans"
          style={{
            fontStyle: 'normal',
            fontWeight: 'bold',
            letterSpacing: 1.4,
          }}>
          Name: {firstName} {lastName}
        </div>
        <div
          className="t--sans"
          style={{
            fontStyle: 'normal',
            fontWeight: 'bold',
            letterSpacing: 1.4,
          }}>
          Dog: {dogName}
        </div>
        <div
          className="t--sans"
          style={{
            fontStyle: 'normal',
            fontWeight: 'bold',
            letterSpacing: 1.4,
          }}>
          Dog ID: {id}
        </div>
        <div
          className="t--sans"
          style={{
            fontStyle: 'normal',
            fontWeight: 'bold',
            letterSpacing: 1.4,
          }}>
          Phone: {phone}
        </div>
        <style jsx>{`
            .result {
              display: block;
              color: inherit;
              font-style: italic;
              border-color: ${GRAY_100};
            }
          `}</style>
      </a>
    </Link>
  );
}
