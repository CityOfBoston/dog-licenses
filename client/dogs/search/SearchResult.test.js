// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import {
  TYPICAL_LICENSE,
  PENDING_LICENSE,
  NO_DATE_LICENSE,
} from '../../../fixtures/client/dog-licenses';

import SearchResult from './SearchResult';

describe('rendering', () => {
  it('renders a typical license', () => {
    expect(
      renderer.create(<SearchResult license={TYPICAL_LICENSE} />).toJSON(),
    ).toMatchSnapshot();
  });

  it('renders a pending license', () => {
    expect(
      renderer.create(<SearchResult license={PENDING_LICENSE} />).toJSON(),
    ).toMatchSnapshot();
  });

  it('renders a license without a death date', () => {
    expect(
      renderer.create(<SearchResult license={NO_DATE_LICENSE} />).toJSON(),
    ).toMatchSnapshot();
  });
});
