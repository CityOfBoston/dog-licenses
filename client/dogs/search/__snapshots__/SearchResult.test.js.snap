// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders a license without a death date 1`] = `
<a
  className="p-a300 br br-t100 b--w result"
  data-jsx={1723197323}
  href="/dogs/license/000002"
  onClick={[Function]}
>
  <div
    className="t--sans"
    data-jsx={1723197323}
    style={
      Object {
        "fontStyle": "normal",
        "fontWeight": "bold",
        "letterSpacing": 1.4,
      }
    }
  >
    Name: 
    Bruce
     
    Banner
  </div>
  <div
    className="t--sans"
    data-jsx={1723197323}
    style={
      Object {
        "fontStyle": "normal",
        "fontWeight": "bold",
        "letterSpacing": 1.4,
      }
    }
  >
    Dog: 
    Poppy
  </div>
  <div
    className="t--sans"
    data-jsx={1723197323}
    style={
      Object {
        "fontStyle": "normal",
        "fontWeight": "bold",
        "letterSpacing": 1.4,
      }
    }
  >
    Dog ID: 
    000002
  </div>
  <div
    className="t--sans"
    data-jsx={1723197323}
    style={
      Object {
        "fontStyle": "normal",
        "fontWeight": "bold",
        "letterSpacing": 1.4,
      }
    }
  >
    Phone: 
    1234567890
  </div>
</a>
`;

exports[`rendering renders a pending license 1`] = `
<a
  className="p-a300 br br-t100 b--w result"
  data-jsx={1723197323}
  href="/dogs/license/000002"
  onClick={[Function]}
>
  <div
    className="t--sans"
    data-jsx={1723197323}
    style={
      Object {
        "fontStyle": "normal",
        "fontWeight": "bold",
        "letterSpacing": 1.4,
      }
    }
  >
    Name: 
    Bruce
     
    Banner
  </div>
  <div
    className="t--sans"
    data-jsx={1723197323}
    style={
      Object {
        "fontStyle": "normal",
        "fontWeight": "bold",
        "letterSpacing": 1.4,
      }
    }
  >
    Dog: 
    Poppy
  </div>
  <div
    className="t--sans"
    data-jsx={1723197323}
    style={
      Object {
        "fontStyle": "normal",
        "fontWeight": "bold",
        "letterSpacing": 1.4,
      }
    }
  >
    Dog ID: 
    000002
  </div>
  <div
    className="t--sans"
    data-jsx={1723197323}
    style={
      Object {
        "fontStyle": "normal",
        "fontWeight": "bold",
        "letterSpacing": 1.4,
      }
    }
  >
    Phone: 
    1234567890
  </div>
</a>
`;

exports[`rendering renders a typical license 1`] = `
<a
  className="p-a300 br br-t100 b--w result"
  data-jsx={1723197323}
  href="/dogs/license/000002"
  onClick={[Function]}
>
  <div
    className="t--sans"
    data-jsx={1723197323}
    style={
      Object {
        "fontStyle": "normal",
        "fontWeight": "bold",
        "letterSpacing": 1.4,
      }
    }
  >
    Name: 
    Bruce
     
    Banner
  </div>
  <div
    className="t--sans"
    data-jsx={1723197323}
    style={
      Object {
        "fontStyle": "normal",
        "fontWeight": "bold",
        "letterSpacing": 1.4,
      }
    }
  >
    Dog: 
    Poppy
  </div>
  <div
    className="t--sans"
    data-jsx={1723197323}
    style={
      Object {
        "fontStyle": "normal",
        "fontWeight": "bold",
        "letterSpacing": 1.4,
      }
    }
  >
    Dog ID: 
    000002
  </div>
  <div
    className="t--sans"
    data-jsx={1723197323}
    style={
      Object {
        "fontStyle": "normal",
        "fontWeight": "bold",
        "letterSpacing": 1.4,
      }
    }
  >
    Phone: 
    1234567890
  </div>
</a>
`;
